daliuge-translator/dlg/translator/tool_commands.py

```diff
@@ -284,11 +284,31 @@ def dlg_submit(parser, args):
     with _open_i(opts.pg_path) as f:
         submit(json.load(f), opts)
 
+
+def cwl(parser, args):
+    print("cwl tool")
+
+    tool.add_logging_options(parser)
+    _add_output_options(parser)
+    apps = _add_unroll_options(parser)
+    _add_partition_options(parser)
+    (opts, args) = parser.parse_args(args)
+    tool.setup_logging(opts)
+
+    # get the pgt
+    pgt = unroll(opts.lg_path, opts.oid_prefix, zerorun=opts.zerorun, app=apps[opts.app])
+    partition(pgt, opts)
```

Comment:
Since the CWL translation takes a physical graph as an input, I'd incline myself to reflect that better in the "dlg cwl" command. What I have in mind is to remove the unroll and partition steps, and let it accept a physical graph as input instead. This in turn makes it composable, so you can run "dlg unroll -L lg.json | dlg cwl" (one could even add a unit test for it, like the one already in `test_tool.py`).

Reply:
Also: ideally it would be nice to follow the same idea of "the output goes directly to stdout by default" that the other commands have, but that would mean more changes to the code below so let's not do that yet. Could we however have a command-line option to specify the output file name? That should be simpler, and takes us a bit further ahead.

Reply:
I like this change too. I'll change dlg cwl to accept a physical graph as input.

I guess we'll json.load() the physical graph file from the opts.pgt_path and then pass it to create_workflow(). But do we have to do anything to the JSON to make it a valid PGT to be passed to create_workflow(), perhaps parse the JSON into a PGT data structure?

Reply:
Mmmmm I don't *think* so, but I might be wrong. Maybe give it a try and see how that goes. If it fails we can have a look at what exactly the requirement is.

Reply:
Yes, it does work. The PGT loaded from JSON is a list of drops/nodes and can be passed directly to create_workflow(). 

But, the PGT that is used in 'lg web' is slightly different. It is retrieved from the pg_mgr using pg_mgr.get_pgt() and the list of drops/nodes is within a 'drops' attribute of the PGT. So I pass pgt.drops to create_workflow() in this case. Maybe the difference is that in this case the structure is actually a PGTP not a PGT.

Reply:
I'll add the ability to specify the output filename.

Reply:
Can now specify the location of the CWL output from "dlg cwl", for example:

`dlg unroll -L test.graph | dlg cwl -o outputs/test.zip`